include/zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -207,8 +208,9 @@ SECTIONS
 #include <zephyr/linker/cplusplus-rom.ld>
 
 	__rodata_region_end = .;
-	MPU_ALIGN(__rodata_region_end -__rom_region_start);
+	MPU_ALIGN(__rodata_region_end - ADDR(rom_start));
```

Comment:
IMO these changes split into another PR to keep the arch changes independent of xmc?

Reply:
Yes, I think that's a good idea. For now, I'd like to keep the XMC4xxx changes here to add some context to the linker changes. I'll split up the PR in a few days.

Reply:
@talih0 possible to split this up?

Reply:
> @talih0 possible to split this up?

I've put these changes in a separate PR.